addons/functions/functions/fnc_spawnTrench.sqf

```diff
@@ -0,0 +1,43 @@
+#include "script_component.hpp"
+/*
+* Author: nomisum
+* Spawns a Trench local to Server
+*
+* Arguments:
+* 0: TrenchClass <STRING>
+* 1: PosDiff <NUMBER 0-1>
+* 2: Pos <ARRAY>
+* 3: VecDirAndUp <NESTED ARRAY>
+*
+* Return Value:
+* None
+*
+* Example:
+* [TrenchObj, 0.5] call grad_trenches_functions_fnc_setTrenchProgress
+*
+* Public: No
+*/
+
+params ["_trenchClass", "_posDiff", "_pos", "_vecDirAndUp"];
+
+//Create a new trench, that is globally visible
+private _trench = createVehicle [_trenchClass, [0,0,0], [], 0, "CAN_COLLIDE"];
+private _digTime = missionNamespace getVariable [getText (configFile >> "CfgVehicles" >> _trenchClass >>"ace_trenches_diggingDuration"), 20];
+
+_trench setObjectTextureGlobal [0, surfaceTexture _pos];
```

Comment:
This should be below the last set, or you will get the Texture from underwater.

Reply:
not in this case as _pos is target pos not current pos

Reply:
We should still move it down to avoide future confusions.

Reply:
ok i dont mind